dwave/cloud/config/models.py

```diff
@@ -0,0 +1,303 @@
+    elif isinstance(headers, abc.Mapping):
+        headers_dict = headers
+    elif isinstance(headers, str):
+        try:
+            # valid  headers = "Field-1: value-1\nField-2: value-2"
+            headers_dict = {key.strip(): val.strip()
+                            for key, val in [line.split(':') for line in
+                                             headers.strip().split('\n')]}
+        except Exception as e:
+            logger.debug("Invalid headers: %r", headers)
+            headers_dict = {}
+    else:
+        raise ValueError("HTTP headers expected in a dict, or a string")
+    logger.debug("parsed headers: %r", headers_dict)
+    config['headers'] = headers_dict
+
+    # parse optional client certificate
+    # (if string, path to ssl client cert file (.pem). If tuple, (‘cert’, ‘key’) pair.)
+    client_cert = config.get('client_cert')
+    client_cert_key = config.get('client_cert_key')
+    if client_cert_key is not None:
+        if client_cert is not None:
+            client_cert = (client_cert, client_cert_key)
+        else:
+            raise ValueError(
+                "Client certificate key given, but the cert is missing")
+    logger.debug("parsed client cert: %r", client_cert)
+    config['cert'] = client_cert
+
+    # parse solver
+    solver = config.get('solver')
+    if isinstance(solver, str):
+        solver = solver.strip()
+    if not solver:
+        solver_def = {}
+    elif isinstance(solver, abc.Mapping):
+        solver_def = solver
+    elif isinstance(solver, str):
+        # support features dict encoded as JSON in our config INI file
+        try:
+            solver_def = json.loads(solver)
+        except Exception:
+            # unparseable (or non-dict) json, assume string name for solver
+            # we'll deprecate this eventually, but for now just convert it to
+            # features dict (equality constraint on full solver name)
+            logger.debug("Invalid solver JSON, assuming string name: %r", solver)
+            solver_def = dict(name__eq=solver)
+        else:
+            # if valid json, has to be a dict
+            if not isinstance(solver_def, abc.Mapping):
+                logger.debug("Non-dict solver JSON, assuming string name: %r", solver)
+                solver_def = dict(name__eq=solver)
+    else:
+        raise ValueError("Expecting a features dictionary or a string name for 'solver'")
+    logger.debug("parsed solver definition: %r", solver_def)
+    config['solver'] = solver_def
+
+    # polling
+    prefix = 'poll_'
+    config['polling_schedule'] = {k[len(prefix):]: v for k, v in config.items()
+                                  if k.startswith(prefix)}
+
+    # retry
+    prefix = 'http_retry_'
+    config['request_retry'] = {k[len(prefix):]: v for k, v in config.items()
+                               if k.startswith(prefix)}
+
+    return ClientConfig.model_validate(config)
+
+
+def dump_config_v1(config: ClientConfig) -> dict:
```

Comment:
This function doesn't seem to be tested anywhere. Should it be?

Reply:
I added just a "smoke test" for now, as we're not using this function (for now), but yeah, I'll add the full test -- the f-string bug above was convincing :smile:

Reply:
Hehe, sounds good.

Reply:
Added tests for `dump_config_v1` in 60b3a9e6d2d573ebd9348176e1c9b66e849a41e0.